File: src/main/java/com/jiujitsustore/service/SupplierService.java
package com.jiujitsustore.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.jiujitsustore.domain.Supplier;
import com.jiujitsustore.dto.SupplierDTO;
import com.jiujitsustore.dto.SupplierSearchDTO;
import com.jiujitsustore.dto.SupplierPageDTO;
import com.jiujitsustore.dto.SupplierConvertCriteriaDTO;
import com.jiujitsustore.service.GenericService;
import com.jiujitsustore.dto.common.RequestDTO;
import com.jiujitsustore.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SupplierService extends GenericService<Supplier, Integer> {

	List<Supplier> findAll();

	ResultDTO addSupplier(SupplierDTO supplierDTO, RequestDTO requestDTO);

	ResultDTO updateSupplier(SupplierDTO supplierDTO, RequestDTO requestDTO);

    Page<Supplier> getAllSuppliers(Pageable pageable);

    Page<Supplier> getAllSuppliers(Specification<Supplier> spec, Pageable pageable);

	ResponseEntity<SupplierPageDTO> getSuppliers(SupplierSearchDTO supplierSearchDTO);
	
	List<SupplierDTO> convertSuppliersToSupplierDTOs(List<Supplier> suppliers, SupplierConvertCriteriaDTO convertCriteria);

	SupplierDTO getSupplierDTOById(Integer supplierId);







}

File: jiujitsustore_ui/src/services/SupplierService.js
import http from "../http-common"; 

class SupplierService {
  getAllSuppliers(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/supplier/suppliers`, searchDTO);
  }

  get(supplierId) {
    return this.getRequest(`/supplier/${supplierId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/supplier?field=${matchData}`, null);
  }

  addSupplier(data) {
    return http.post("/supplier/addSupplier", data);
  }

  update(data) {
  	return http.post("/supplier/updateSupplier", data);
  }
  
  uploadImage(data,supplierId) {
  	return http.postForm("/supplier/uploadImage/"+supplierId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SupplierService();
